qemu/tests/win_virtio_serial_data_transfer.py

```diff
@@ -0,0 +1,205 @@
+        for script in guest_scripts.split(";"):
+            src_file = os.path.join(deps_dir, script)
+            self.vm.copy_files_to(src_file, "c:\\", timeout=60)
+        self.event = "prepare"
+
+    @error.context_aware
+    def uninstall_required_scripts(self):
+        """ cleanup installed script in guest """
+        guest_scripts = self.params["guest_scripts"]
+        error.context("Copy test scripts to guest.", logging.info)
+        session = self._get_session()
+        try:
+            for script in guest_scripts.split(";"):
+                script = "c:\\%s" % script
+                session.cmd("del /F %s" % script, timeout=60)
+            self.event = "cleanup"
+        except Exception:
+            logging.warn("Failed to remove %s" % guest_scripts)
+        finally:
+            session.close()
+
+
+def run(test, params, env):
+    """
+    QEMU 'Windows virtio-serial data transfer' test
+
+    1) Start guest with one virtio-serial-pci and two virtio-serial-port.
+    2) Make sure vioser.sys verifier enabled in guest.
+    3) Transfering data from host to guest via virtio-serial-port in a loop.
+    4) Reboot guest.
+    5) Repeat step 3.
+    6) Reboot guest by system_reset qmp command.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def logtime_transfer(win_serial_test, data_file, repeat):
+        """
+        Run data transfer from host to guest in loop
+        """
+        for count in xrange(1, repeat + 1):
+            logging.info("transfer repeat %d times" % count)
+            win_serial_test.transfer_data(data_file)
+
+    transfer_timeout = int(params.get("transfer_timeout", 900))
+    repeat = int(params.get("repeat_times", 1))
+    deps_dir = data_dir.get_deps_dir("win_serial")
+    send_script = params.get("host_send_script", "serial-host-send.py")
+    data_file = os.path.join(deps_dir, send_script)
+    interrupt = params.get("interrupt", "yes") == "yes"
+    with WinSerialTransferTest(test, params, env) as win_serial_test:
+        args = (win_serial_test, data_file, repeat)
+        transfer_thread = utils.InterruptedThread(logtime_transfer, args)
+        transfer_thread.start()
+        start_transfer = utils_misc.wait_for(
+            lambda: win_serial_test.event == "sent",
+            timeout=transfer_timeout)
+        if not start_transfer:
+            raise error.TestFail("No transimiation start "
+                                 "in last '%s' seconds" % transfer_timeout)
+        if interrupt:
+            sub_step = params.get("sub_step", "reboot")
+            getattr(win_serial_test.vm, sub_step)()
+            if sub_step == "shutdown":
+                win_serial_test.vm.create()
+        elif params.get("sub_step") == "pause_resume":
+            win_serial_test.vm.pause()
+            pause_time = float(params.get("pause_time", 300))
```

Comment:
it seems if `sub_step = shutdown`, `win_serial_test.vm` does not have an attribute named `shutdown`. Available choices are `graceful_shutdown` or `wait_for_shutdown`. Please change here or the cfg file.